Framework/MDAlgorithms/src/FindPeaksMD.cpp

```diff
@@ -322,6 +325,8 @@ void FindPeaksMD::findPeaks(typename MDEventWorkspace<MDE, nd>::sptr ws) {
   for (uint16_t iexp = 0; iexp < ws->getNumExperimentInfo(); iexp++) {
     ExperimentInfo_sptr ei = ws->getExperimentInfo(iexp);
     this->readExperimentInfo(ei, boost::dynamic_pointer_cast<IMDWorkspace>(ws));
+
+    Geometry::InstrumentRayTracer tracer(inst);
```

Comment:
This is (one of?) the changes for better performance, caching the ray tracer?

`inst` does not look specific to each loop iteration, why not move outside? Is it being updated by `readExperimentInfo`?

Reply:
function `readExperimentInfo` Line 327 actually resets the `inst`. Potentially every experiment info in the multiple experiment info has a different instrument associated.